erpnext_germany/erpnext_germany/doctype/business_trip_region/business_trip_region.py

```diff
@@ -21,4 +22,91 @@ class BusinessTripRegion(Document):
 		valid_from: DF.Date | None
 		whole_day: DF.Currency
 	# end: auto-generated types
-	pass
+
+	def validate(self):
+		"""Main validation called automatically before save."""
+		self._ensure_title()
+		self._warn_if_no_allowances()
+		self._validate_allowance_ranges()
+
+	def _ensure_title(self):
+		"""Auto-fill title if not set."""
+		if not self.title:
+			self.title = f"Region - {self.name or 'New'}"
```

Comment:
This is unrelated to this PR. Better to keep it out and maybe create a separate PR.